modules/utxo-lib/src/bitgo/signature.ts

```diff
@@ -560,6 +562,64 @@ export function verifySignature(
   return signatureVerifications.length > 0 && signatureVerifications.every((v) => v.signedBy !== undefined);
 }
 
+/**
+ * @param v
+ * @param publicKey
+ * @return true iff signature is by publicKey (or xonly variant of publicKey)
+ */
+function isSignatureByPublicKey(v: SignatureVerification, publicKey: Buffer): boolean {
+  return (
+    !!v.signedBy &&
+    (v.signedBy.equals(publicKey) ||
+      /* for p2tr signatures, we pass the pubkey in 33-byte format recover it from the signature in 32-byte format */
+      (publicKey.length === 33 && isSignatureByPublicKey(v, publicKey.slice(1))))
+  );
+}
+
+/**
+ * @param transaction
+ * @param inputIndex
+ * @param prevOutputs - transaction outputs for inputs
+ * @param publicKeys - public keys to check signatures for
+ * @return array with signature verification result for each public key in publicKeys
```

Comment:
Minor note but I think still helpful in making this all easy to follow, I'd say something more like "boolean array indicating whether a valid signature exists for each public key in publicKeys" - this as opposed to "signature verification result" since we have a `SignatureVerification` type in use that's distinct from what this method is returning.

Reply:
improved docstring